pkg/segment/query/processor/fillnullcommand_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package processor
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/siglens/siglens/pkg/segment/query/iqr"
+	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+	toputils "github.com/siglens/siglens/pkg/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func insertColumnsWithSomeNulls(t *testing.T, iqr *iqr.IQR, fillValue utils.CValueEnclosure, columnsToInsert []string, valuesCountToInsert int) (
+	map[string][]utils.CValueEnclosure, map[string][]utils.CValueEnclosure) {
+	knownValuesToInsert := make(map[string][]utils.CValueEnclosure)
+	knownValuesToExpect := make(map[string][]utils.CValueEnclosure)
+
+	var err error
+
+	for _, column := range columnsToInsert {
+		valuesToInsert := make([]utils.CValueEnclosure, valuesCountToInsert)
+		valuesToExpect := make([]utils.CValueEnclosure, valuesCountToInsert)
+		for i := 0; i < valuesCountToInsert; i++ {
+			insertNull := i%2 == 0
+
+			cValueToInsert := utils.CValueEnclosure{}
+			var cValueToExpect utils.CValueEnclosure
+			if insertNull {
+				err = cValueToInsert.ConvertValue(nil)
+				assert.NoError(t, err)
+				cValueToExpect = fillValue
+			} else {
+				err = cValueToInsert.ConvertValue(fmt.Sprintf("value-%v", i))
+				assert.NoError(t, err)
+				cValueToExpect = cValueToInsert
+			}
+
+			valuesToInsert[i] = cValueToInsert
+			valuesToExpect[i] = cValueToExpect
+		}
+
+		knownValuesToInsert[column] = valuesToInsert
+		knownValuesToExpect[column] = valuesToExpect
+	}
+
+	err = iqr.AppendKnownValues(knownValuesToInsert)
+	assert.NoError(t, err)
+
+	return knownValuesToInsert, knownValuesToExpect
+}
+
+func Test_fillNullCommandWithFields(t *testing.T) {
```

Comment:
nit: I found this unit test a little hard to understand. I think it's correct, so you can keep it as is, but FYI this is what I would have done:
1. Make the fillnull processor like you did
2. Hard code some IQRs with just NewIQR and then AppendKnownValues, passing in a hard coded map so it's really obvious what the IQR has.
3. Call Process() like you did to get an IQR
4. Use ReadColumn or ReadAllColumns on the IQR, and hard code what you expect for each value.

For example with the [sort command tests](https://github.com/siglens/siglens/blob/develop/pkg/segment/query/processor/sortcommand_test.go#L30) I did basically that, and I think that makes it really obvious that the asserts in the test are correct, and makes it obvious which edge cases I'm testing. In general for me personally, I'm much more lenient towards hardcoding and duplicating code in unit tests than I am in "real code" because I want unit tests to be obvious

Reply:
Got it, next Time, I will just hard code some values.